### ISSUE TITLE
Calling a method on an autoload instance that was `queue_freed` causes a crash

### ISSUE THREAD

**Issue by darthdeus** [godotengine/godot#79591]
### Godot version

4.1.1.stable [bd6af8e0e]

### System information

Godot v4.1.1.stable (bd6af8e0e) - Arch Linux #1 SMP PREEMPT_DYNAMIC Thu, 06 Jul 2023 13:07:19 +0000 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (nvidia) - AMD Ryzen 9 3900X 12-Core Processor (24 Threads)

### Issue description

Calling a method on an autoload that was freed crashes with the following stacktrace

I reproduced this on the official 4.1 release, then on both 4.1 and 4.1.1 custom debug builds.

```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.1.stable.custom_build (bd6af8e0ea69167dd0627f3bd54f9105bda0f8b5)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /usr/lib/libc.so.6(+0x39ab0) [0x7fb31f452ab0] (??:0)
[2] /home/darth/projects/godot/bin/godot.linuxbsd.editor.dev.x86_64(__dynamic_cast+0x100) [0x55afc217b3e0] (??:?)
[3] GDScriptNativeClass* Object::cast_to<GDScriptNativeClass>(Object*) (/home/darth/projects/godot/./core/object/object.h:754)
[4] GDScriptLanguage::debug_get_globals(List<String, DefaultAllocator>*, List<Variant, DefaultAllocator>*, int, int) (/home/darth/projects/godot/modules/gdscript/gdscript_editor.cpp:377 (discriminator 1))
[5] RemoteDebugger::debug(bool, bool) (/home/darth/projects/godot/core/debugger/remote_debugger.cpp:449)
[6] ScriptDebugger::debug(ScriptLanguage*, bool, bool) (/home/darth/projects/godot/core/debugger/script_debugger.cpp:96)
[7] GDScriptLanguage::debug_break(String const&, bool) (/home/darth/projects/godot/modules/gdscript/gdscript_editor.cpp:245)
[8] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (/home/darth/projects/godot/modules/gdscript/gdscript_vm.cpp:3616 (discriminator 1))
[9] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/darth/projects/godot/modules/gdscript/gdscript.cpp:1892 (discriminator 1))
[10] bool Node::_gdvirtual__process_call<false>(double) (/home/darth/projects/godot/scene/main/node.h:318 (discriminator 6))
[11] Node::_notification(int) (/home/darth/projects/godot/scene/main/node.cpp:57)
[12] Node::_notificationv(int, bool) (/home/darth/projects/godot/./scene/main/node.h:49 (discriminator 14))
[13] CanvasItem::_notificationv(int, bool) (/home/darth/projects/godot/./scene/main/canvas_item.h:45 (discriminator 3))
[14] Node2D::_notificationv(int, bool) (/home/darth/projects/godot/./scene/2d/node_2d.h:37 (discriminator 3))
[15] Object::notification(int, bool) (/home/darth/projects/godot/core/object/object.cpp:798)
[16] SceneTree::_process_group(SceneTree::ProcessGroup*, bool) (/home/darth/projects/godot/scene/main/scene_tree.cpp:946)
[17] SceneTree::_process(bool) (/home/darth/projects/godot/scene/main/scene_tree.cpp:1018 (discriminator 2))
[18] SceneTree::process(double) (/home/darth/projects/godot/scene/main/scene_tree.cpp:510)
[19] Main::iteration() (/home/darth/projects/godot/main/main.cpp:3425 (discriminator 3))
[20] OS_LinuxBSD::run() (/home/darth/projects/godot/platform/linuxbsd/os_linuxbsd.cpp:912 (discriminator 1))
[21] /home/darth/projects/godot/bin/godot.linuxbsd.editor.dev.x86_64(main+0x19b) [0x55afbcc73234] (/home/darth/projects/godot/platform/linuxbsd/godot_linuxbsd.cpp:76)
[22] /usr/lib/libc.so.6(+0x23850) [0x7fb31f43c850] (??:0)
[23] /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7fb31f43c90a] (??:0)
[24] /home/darth/projects/godot/bin/godot.linuxbsd.editor.dev.x86_64(_start+0x25) [0x55afbcc72fc5] (??:?)
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

Load the example project, press enter twice

To reproduce manually simply create an autoload e.g. `Global.gd` mapped to `Global` autoload as follows

```
extends Node

func foo():
	print("foo")
	queue_free()
```

and then call it twice, e.g.

```
func _process(_delta: float) -> void:
	if Input.is_action_just_pressed("ui_accept"):
		Global.foo()
```

### Minimal reproduction project

[repr.zip](https://github.com/godotengine/godot/files/12073870/repr.zip)

**Comment by AcatXIo** [comment]
I could not reproduce the crash with 4.2.dev3. The only thing that happened is the following error was printed:
```txt
Attempt to call function 'foo' in base 'previously freed' on a null instance.
```

**Comment by dalexeev** [comment]
* I can't reproduce this in the current master. This is probably a duplicate of #71793, which was fixed by #73679.

Thanks for the contribution nonetheless!
